lib/gc/Transforms/FoldTensorOperation.cpp

```diff
@@ -0,0 +1,53 @@
+//===-- FoldTensorOperation.cpp - fold tensor op ----------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "gc/Transforms/Passes.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Tensor/Transforms/Transforms.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/Support/Casting.h"
+
+namespace mlir {
+namespace gc {
+
+#define GEN_PASS_DEF_FOLDTENSOROPERATION
+#include "gc/Transforms/Passes.h.inc"
+namespace {
+
+/// LowerAndFoldTensorOperation is a pass that fold some useless tensor
+/// operation.
+struct FoldTensorOperationPass
+    : public impl::FoldTensorOperationBase<FoldTensorOperationPass> {
```

Comment:
The struct name `FoldTensorOperationPass` is not consistent with the comment `LowerAndFoldTensorOperation`.